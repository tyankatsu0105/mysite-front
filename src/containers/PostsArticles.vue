<template>
	<div class="PostsArticles">
		<PostsArticle
			v-for="edge in page.allContentfulPosts.edges"
			:key="edge.node.id"
			:article="edge.node"
		/>
	</div>
</template>

<script>
import PostsArticle from "@/components/PostsArticle.vue";
export default {
  components: {
    PostsArticle
  },
  props: {
    page: {
      type: Object,
      default: () => {}
    }
  }
};
</script>

<style scoped lang="scss">
.PostsArticles {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-auto-rows: auto;
  grid-gap: 30px;
  justify-items: center;

  @media screen and (max-width: 1280px) {
    grid-template-columns: 1fr 1fr;
  }

  @include mq-xs {
    grid-template: 1fr / 100%;
  }
}
</style>
