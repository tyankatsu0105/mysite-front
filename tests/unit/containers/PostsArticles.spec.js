import { shallowMount } from "@vue/test-utils";
import PostsArticles from "@/containers/PostsArticles.vue";

const propsData = {
  page: {
    allContentfulPosts: {
      pageInfo: {
        totalPages: 4,
        currentPage: 1
      },
      edges: [
        {
          node: {
            id: "56X3zCipcA2Gw8WOGgSico",
            name: "Title",
            eyecatch: ["4bRUr9soZOiy6QwukMsqeg"],
            contents: "contents",
            title: "タイトル２",
            slug: "fuga",
            path: "/posts/fuga",
            date: "2019.01.05"
          }
        },
        {
          node: {
            id: "gsSl9UAyukmMYCKaEgwsW",
            name: "Title",
            eyecatch: ["4bRUr9soZOiy6QwukMsqeg"],
            contents: "contents",
            title: "タイトル２",
            slug: "fuga",
            path: "/posts/fuga",
            date: "2019.01.05"
          }
        }
      ]
    }
  }
};

describe("PostsArticles", () => {
  it("render", () => {
    const wrapper = shallowMount(PostsArticles, {
      propsData,
      stubs: ["g-link", "g-image", "font-awesome-icon"]
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
