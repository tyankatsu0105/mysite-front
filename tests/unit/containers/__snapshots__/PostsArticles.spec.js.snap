// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsArticles render 1`] = `
<div class="PostsArticles">
  <postsarticle-stub article="[object Object]"></postsarticle-stub>
  <postsarticle-stub article="[object Object]"></postsarticle-stub>
</div>
`;
